code/controllers/subsystems/processing/processing.dm

```diff
@@ -7,33 +7,50 @@ SUBSYSTEM_DEF(processing)
 	wait = 1 SECOND
 
 	var/list/processing = list()
-	var/list/current_run = list()
 	var/process_proc = /datum/proc/Process
 
 	var/debug_last_thing
 	var/debug_original_process_proc // initial() does not work with procs
+	var/nextProcessingListPosition = 0 //position of next thing to be processed, inside of currently processed list
+
+/datum/controller/subsystem/processing/stopProcessingWrapper(var/D) //called before a thing stops being processed
+	if (!nextProcessingListPosition) //0 position means currently processed list is not paused or running, no point in adjusting last position due to removals from list
+		return
+	var/position = processing.Find(D) //find exact position in list
```

Comment:
Does this have potential be a net loss in performance? For example, if a big explosion causes many items to be deleted at once?

Reply:
The deletions would have to happen during a process call or between rescheduling. Otherwise the proc returns at start due to no process running. It is unlikely to affect performance.